protocol/wallet.re

```diff
@@ -1,29 +1,66 @@
-open Helpers;
 open Mirage_crypto_ec;
 
-[@deriving (ord, yojson)]
-type t = BLAKE2B.t;
+type t = Ed25519.priv;
+type pub_ = Ed25519.pub_;
 
-let of_address = pubkey => {
-  let to_yojson = [%to_yojson: Address.t];
-  pubkey |> to_yojson |> Yojson.Safe.to_string |> BLAKE2B.hash;
-};
-let pubkey_matches_wallet = (key, wallet) => {
-  of_address(key) == wallet;
-};
-let get_pub_key = Ed25519.pub_of_priv;
+let wallet_of_privkey = priv => priv;
+let wallet_to_privkey = priv => priv;
 
-let make_address = () => {
-  let (_key, pub_) = Ed25519.generate();
-  pub_ |> of_address;
-};
-let make_wallet = () => {
-  let (key, pub_) = Ed25519.generate();
-  let wallet_address = of_address(pub_);
+let pub_of_Ed25519pub = pub_ => pub_;
+let pub_to_Ed25519pub = pub_ => pub_;
+
+let pubkey_of_wallet = Ed25519.pub_of_priv;
+
+let make_pair = () => Ed25519.generate();
+let make_pubkey = () => make_pair() |> snd;
 
-  (key, wallet_address);
+let to_hex = str => {
+  let `Hex(str) = Hex.of_string(str);
+  str;
 };
-let address_to_blake = t => t;
-let address_of_blake = t => t;
-let address_to_string = wallet =>
-  wallet |> address_to_blake |> BLAKE2B.to_string;
+let of_hex = hex => Hex.to_string(`Hex(hex));
+
+let to_yojson = t =>
+  `String(Ed25519.priv_to_cstruct(t) |> Cstruct.to_string |> to_hex);
+let of_yojson =
+  fun
+  | `String(key) =>
+    // TODO: this raises an exception
```

Comment:
This is probably an outdated comment

Reply:
Thought so too, just didn't remove it to keep this PR small https://github.com/marigold-dev/sidechain/blob/main/protocol/address.re#L16